lib/oppen/scan_stack.rb

```diff
@@ -9,23 +9,36 @@ class ScanStack
     extend Mixins
 
     def initialize(size, config)
+      # Points to the bottom of the stack.
       @bottom = 0
+      # Printing config.
       @config = config
+      # Emptiness flag.
       @empty = true
-      @stack = Array.new(size)
+      # The fixed sized stack.
+      @stack = Array.new size
+      # Points to the top of the stack.
       @top = 0
     end
```

Comment:
Since these are not doc comments, we can just do something like this:

```
      @stack = Array.new size # The fixed sized stack.
      @top = 0                # Points to the top of the stack.   
```

etc. Compact and readable.